### PR TITLE
[Impeller] Fix Vulkan validation error when there is a blit directed at a swapchain image.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This shows up in the the aiks test harness in the color wheel with advanced blends.

Fixes https://github.com/flutter/flutter/issues/129853

### CODE REVIEW

**Comment by chinmaygarde** [comment]
I thought we were running unit-tests with validation enabled but that may not the case for test that toss open a playground window. And this one was in an advanced blend that had to be manually selected.

File: impeller/renderer/backend/vulkan/allocator_vk.cc [hunk 1]
```diff
@@ -164,8 +164,8 @@ static constexpr vk::ImageUsageFlags ToVKImageUsageFlags(
   }
 
   if (mode != StorageMode::kDeviceTransient) {
-    // TODO (https://github.com/flutter/flutter/issues/121634):
-    // Add transfer usage flags to support blit passes
+    // Add transfer usage flags to support blit passes only if image isn't
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/allocator_vk.cc:167]: I found this outdated comment as I was looking for image usage flags. These flags are for allocated texture sources. The issue was with images that were wrapped swapchain images.

**Review by bdero — APPROVED** [review]

**Comment by jonahwilliams** [comment]
Does this fix https://github.com/flutter/flutter/issues/129853 ?

**Comment by chinmaygarde** [comment]
It fixes that exact issue yes. But this has thrown up another issue (a write-after-write hazard on the blit yielded by `vk::ValidationFeatureEnableEXT::eSynchronizationValidation`). I am chasing that one down now. But this is a sound fix. We should be able to blit to swapchain images.

**Pull request merged**